docs/scripts_md/DCMSUM.md

```diff
@@ -0,0 +1,226 @@
+file.
+
+INPUT: list of DICOM files
+
+RETURNS: array of hashes with acquisition parameters for each file
+
+### collapse($self->{acqu\_AoH})
+
+Collapses the AoH to get a summary of acquisitions.
+
+INPUT: array of hashes with acquisition parameters for each DICOM file
+
+RETURNS: hash table acquisition summary collapsed by unique acquisition
+definitions
+
+### acquisitions(self->{acqu\_Sum})
+
+Sorts the Array of Hash by acquisition number.
+
+INPUT: array of hash table acquisition summary
+
+RETURNS: acquisition listing sorted by acquisition number to be used for summary
+
+### content\_list($dcmdir)
+
+Gets DICOM info from all files in a directory.
+
+Note: I added the -k5 on August 28th 2006 because the guys in Kupio assign
+duplicate FN SN EN values for scouts and subsequent scans.
+
+INPUT: DICOM directory
+
+RETURNS: sorted DICOM information from all the files in the DICOM directory
+
+### read\_dicom\_data($file)
+
+Gets DICOM info from a DICOM file.
+
+INPUT: DICOM file to read
+
+RETURNS: array with pertinent DICOM information read from the DICOM file
+
+### fill\_header()
+
+Fills header information reading the first valid DICOM file.
+
+RETURNS: header information
+
+### confirm\_single\_study()
+
+Confirms that only one DICOM study is in the DICOM directory to be archived.
+Returns False if there is more than one StudyUID, otherwise it returns that
+StudyUID.
+
+RETURNS: StudyUID found in the DICOM directory, or false if more than one
+study was found
+
+### print\_header()
+
+Prints HEADER using the format defined in `&format_head` function.
+
+### format\_head()
+
+Format definition to print the head of the HEADER.
+
+### print\_content()
+
+Prints the CONTENT using formats defined in `&write_content_head`.
+
+### write\_content\_head()
+
+Prints the Content head.
+
+### write\_dcm($dcm)
+
+Prints all DICOM files.
+
+INPUT: array of DICOM information to print
+
+### write\_other($dcm)
+
+Prints all other files.
+
+INPUT: array of other files information
+
+### print\_acquisitions()
+
+Prints Acquisitions using formats from `&write_acqu_head` and
+`&write_acqu_content`
+
+### write\_acqu\_head()
+
+Prints acquisition table header.
+
+### write\_acqu\_content($acqu)
+
+Prints acquisition's content.
+
+INPUT: array of acquisition information to print
+
+### print\_footer()
+
+Prints footer using formats from `&write_footer`.
+
+### write\_footer()
+
+Prints footer summary information.
+
+### dcmsummary()
+
+Prints the whole thing using `&print_header`, `&print_content`,
+`&print_acquisitions` and `&print_footer`. This is what you really want.
+
+**Unrelated but useful functions**
+
+### trimwhitespace($string)
+
+Gets rid of nasty whitespace
+
+INPUT: string to remove white space
+
+RETURNS: string without white space
+
+### date\_format($first, $second)
+
+Pass it a date in YYYYMMDD and you get YYYY-MM-DD.
+Pass it two of these and you get the difference in decimal and Y M +/- Days.
```

Comment:
I find the sentence structure a bit weird.
Perhaps maintain the structure above: like saying for example: A function that converts YYYMMMDD to YYYY-MM-DD